src/webparts/todolist/ToDoListWebPart.ts

```diff
@@ -3,106 +3,44 @@ import * as ReactDom from "react-dom";
 import { Version } from "@microsoft/sp-core-library";
 import {
   IPropertyPaneConfiguration,
-  PropertyPaneTextField
+  PropertyPaneDropdown
 } from "@microsoft/sp-property-pane";
 import { BaseClientSideWebPart } from "@microsoft/sp-webpart-base";
-import { IReadonlyTheme } from "@microsoft/sp-component-base";
 
 import * as strings from "ToDoListWebPartStrings";
 import ToDoList from "./components/ToDoList";
 import { IToDoListProps } from "../../interfaces";
+import { spfi, SPFI, SPFx } from "@pnp/sp";
+import "@pnp/sp/webs";
+import "@pnp/sp/items";
+import "@pnp/sp/lists";
 
 export interface IToDoListWebPartProps {
-  description: string;
+  listName: string;
 }
 
 export default class ToDoListWebPart extends BaseClientSideWebPart<IToDoListWebPartProps> {
-  private _isDarkTheme: boolean = false;
-  private _environmentMessage: string = "";
+  private _sp: SPFI;
 
   public render(): void {
     const element: React.ReactElement<IToDoListProps> = React.createElement(
       ToDoList,
       {
-        websiteUrl: this.context.pageContext.web.absoluteUrl,
-        spHttpClient: this.context.spHttpClient,
-        isDarkTheme: this._isDarkTheme,
-        environmentMessage: this._environmentMessage,
-        hasTeamsContext: !!this.context.sdks.microsoftTeams,
         userDisplayName: this.context.pageContext.user.displayName,
-        context: this.context
+        sp: this._sp,
+        listName: this.properties.listName
       }
     );
     ReactDom.render(element, this.domElement);
   }
 
-  protected onInit(): Promise<void> {
-    return this._getEnvironmentMessage().then(message => {
-      this._environmentMessage = message;
-    });
+  protected async onInit(): Promise<void> {
+    await super.onInit();
+    this._sp = spfi().using(SPFx(this.context));
```

Comment:
Will be better to make it using service like here https://dev.azure.com/engagesq/_git/Engage%20Squared%20-%20Intranet%20Components?path=/components/content-author-dashboard/src/services/ListService.ts. But for small webparts this initialization also fine